packages/hardhat-core/src/internal/hardhat-network/provider/modules/hardhat.ts

```diff
@@ -180,6 +201,72 @@ export class HardhatModule {
     return true;
   }
 
+  // hardhat_setBalance
+
+  private _setBalanceParams(params: any[]): [Buffer, BN] {
+    return validateParams(params, rpcAddress, rpcQuantity);
+  }
+
+  private async _setBalanceAction(address: Buffer, newBalance: BN) {
+    await this._node.setAccountBalance(new Address(address), newBalance);
+    return true;
+  }
+
+  // hardhat_setCode
+
+  private _setCodeParams(params: any[]): [Buffer, Buffer] {
+    return validateParams(params, rpcAddress, rpcData);
+  }
+
+  private async _setCodeAction(address: Buffer, newCode: Buffer) {
+    await this._node.setAccountCode(new Address(address), newCode);
+    return true;
+  }
+
+  // hardhat_setNonce
+
+  private _setNonceParams(params: any[]): [Buffer, BN] {
+    return validateParams(params, rpcAddress, rpcQuantity);
+  }
+
+  private async _setNonceAction(address: Buffer, newNonce: BN) {
+    await this._node.setAccountNonce(new Address(address), newNonce);
+    return true;
+  }
+
+  // hardhat_setStorageSlot
+
+  private _setStorageSlotParams(params: any[]): [Buffer, BN, Buffer] {
+    const validatedParams = validateParams(
+      params,
+      rpcAddress,
+      rpcQuantity,
+      rpcData
+    );
+
+    const MAX_WORD_VALUE = new BN(2).pow(new BN(256));
+    if (validatedParams[1].gt(MAX_WORD_VALUE)) {
+      throw new InvalidInputError("Storage key must not be greater than 2^256");
```

Comment:
We should show the value used in the error message.

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1435/commits/6726019c4d478e7835219a7f402580f16138d319